File: go/arraysandhashing/twosum.go
package arraysandhashing

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)

	for i, num := range nums {
		v, ok := m[target-num]
		if ok {
			return []int{v, i}
		}
		m[num] = i
	}

	return []int{}
}

File: ts/src/arrays-and-hashing/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
  const map = new Map<number, number>();

  for (let i = 0; i < nums.length; i++) {
    const num = nums[i];
    const j = map.get(num);
    if (j !== undefined) {
      return [j, i];
    }

    map.set(target - num, i);
  }

  return [];
}

export default twoSum;
